pod-operation/src/state_machine.rs

```diff
@@ -226,21 +248,36 @@ impl StateMachine {
 	/// Perform operations when the pod is loading
 	fn _load_periodic(&mut self) -> State {
 		info!("Rolling Load state");
-
 		State::Load
 	}
 
 	/// Perform operations when the pod is running
 	fn _running_periodic(&mut self) -> State {
 		info!("Rolling Running state");
-		let encoder_value = self.wheel_encoder.measure().expect("wheel encoder faulted"); // Read the encoder value
-		if encoder_value > STOP_THRESHOLD {
+
+		let encoder_value = self.wheel_encoder.lock().unwrap();
+		let distance = encoder_value.get_distance();
+		let velocity: f32 = encoder_value.get_velocity();
+
+		let full_json = json!({
+			"distance": distance,
+			"velocity": velocity,
+		});
```

Comment:
Chore: we'll need to nest the values inside another property called `wheel_encoder` as we did before